trade-finance-manager-ui/server/controllers/utilisation-reports/add-payment/index.ts

```diff
@@ -87,21 +99,27 @@ export const addPayment = async (req: AddPaymentRequest, res: Response) => {
     const { reportId } = req.params;
     const checkedCheckboxIds = getPremiumPaymentsCheckboxIdsFromObjectKeys(req.body);
     const feeRecordIds = checkedCheckboxIds.map((checkboxId) => getFeeRecordIdFromPremiumPaymentsCheckboxId(checkboxId));
-    const { errors, formValues, paymentNumber } = extractAddPaymentFormValuesAndValidateIfPresent(req.body);
+    const feeRecordPaymentCurrency = getFeeRecordPaymentCurrencyFromPremiumPaymentsCheckboxId(checkedCheckboxIds[0]);
+
+    const { isAddingPayment, errors, formValues, paymentNumber } = extractAddPaymentFormValuesAndValidateIfPresent(req.body, feeRecordPaymentCurrency);
+    const canSubmitFormValues = errors.errorSummary.length === 0;
 
-    if ('addPaymentFormSubmission' in req.body && errors.errorSummary.length === 0) {
-      // TODO FN-1739 save data upon valid submission
-      return res.status(501).send('Your request was valid, but we have not finished building the feature');
+    if (isAddingPayment && canSubmitFormValues) {
+      const { addAnotherPayment, ...paymentFormValues } = formValues;
+      await api.addPaymentToFeeRecords(reportId, paymentFormValues, feeRecordIds, user, userToken);
+      if (addAnotherPayment && addAnotherPayment !== 'true') {
+        return res.redirect(`/utilisation-reports/${reportId}`);
+      }
     }
 
-    const selectedFeeRecordDetails = await api.getSelectedFeeRecordsDetails(Number(reportId), feeRecordIds, userToken);
+    const selectedFeeRecordDetails = await api.getSelectedFeeRecordsDetails(reportId, feeRecordIds, userToken);
     return renderAddPaymentPage(res, {
       user,
       activePrimaryNavigation: PRIMARY_NAVIGATION_KEYS.UTILISATION_REPORTS,
-      reportId: Number(reportId),
+      reportId,
       selectedFeeRecordCheckboxIds: checkedCheckboxIds,
       errors,
-      formValues,
+      formValues: canSubmitFormValues ? EMPTY_ADD_PAYMENT_FORM_VALUES : formValues,
```

Comment:
What's the scenario for this again? is this supporting clearing the page if we are adding a second payment?

Reply:
Exactly - without this, the second payment form would be pre-populated with the data from the first payment